docs/content/en/highlights/2021-07-13-helm-customConfig.md

```diff
@@ -0,0 +1,324 @@
++    ...
+```
+
+### vectorSink
+
+```diff title="values.yaml"
+   vectorSink:
++    enabled: false
+     sinkId: vector_sink
+     inputs: ["kubernetes_logs"]
+     host: vector
+     port: "9000"
+     config: {}
+     rawConfig: null
+   ...
++  customConfig:
++    ...
++    sinks:
++      ...
++      vector_sink:
++        type: vector
++        inputs: ["kubernetes_logs"]
++        address: vector:9000
++    ...
+```
+
+### internalMetricsSource
+
+```diff title="values.yaml"
+   internalMetricsSource:
++    enabled: false
+     sourceId: internal_metrics
+     config: {}
+     rawConfig: null
+   ...
++  customConfig:
++    ...
++    sources:
++      ...
++      internal_metrics:
++        type: internal_metrics
++    ...
+```
+
+### hostMetricsSource
+
+Users should set the PROCFS_ROOT and SYSFS_ROOT environment variables, as well as mounting the
+required hostPaths.
+
+```diff title="values.yaml"
+   hostMetricsSource:
++    enabled: false
+     sourceId: host_metrics
+     config:
+       filesystem:
+         devices:
+           excludes: [binfmt_misc]
+         filesystems:
+           excludes: [binfmt_misc]
+         mountpoints:
+           excludes: ["*/proc/sys/fs/binfmt_misc"]
+     rawConfig: null
+   ...
++  customConfig:
++    ...
++    sources:
++      ...
++      host_metrics:
++        type: host_metrics
++        filesystem:
++          devices:
++            excludes: ["binfmt_misc"]
++          filesystems:
++            excludes: ["binfmt_misc"]
++          mountPoints:
++            excludes: ["*/proc/sys/fs/binfmt_misc"]
++  env:
++    - name: PROCFS_ROOT
++      value: /host/proc
++    - name: SYSFS_ROOT
++      value: /host/sys
++  extraVolumeMounts:
++    - name: procfs
++      mountPath: /host/proc
++      readOnly: true
++    - name: sysfs
++      mountPath: /host/sys
++      readOnly: true
++  extraVolumes:
++    - name: procfs
++      hostPath:
++        path: /proc
++    - name: sysfs
++      hostPath:
++        path: /sys
++    ...
+```
+
+### prometheusSink
+
+The `extraContainerPorts` or `service` key should be used to expose the port configured in `customConfig`.
+
+The `prometheusSink.podMonitor` key has been moved to a top level key and can be accessed directly at
+`podMonitor`. The `addPodAnnotations` option has been removed in favor setting the required annotations
+with the `podAnnotations` key.
+
+```diff title="values.yaml"
+   prometheusSink:
++    enabled: false
+     sinkId: prometheus_sink
+     inputs: []
+     excludeInternalMetrics: false
+     listenAddress: "0.0.0.0"
+     listenPort: "9090"
+     config: {}
+     rawConfig: null
+   ...
++  customConfig:
++    ...
++    sinks:
++      ...
++      prometheus_sink:
++        type: prometheus_exporter
++        inputs: ["host_metrics", "internal_metrics"]
++        address: 0.0.0.0:9090
++    ...
+```
+
+## Using `customConfig`
+
+We've configured the ConfigMap template to `fail` if the deprecated keys are enabled at the same time
```

Comment:
```suggestion
We've configured the ConfigMap template to fail if the deprecated keys are enabled at the same time
```

Ignore me if fail is a status or something similar.

Reply:
This one is interesting because we're calling a function called `fail`, which errors the install/template with an error message - it definitely reads weirder with how I phrased/identified it